Automattic/wp-calypso
# Fixes webview integration with the Jetpack mobile app

Related to SELFSERVE-519

## Proposed Changes

This PR make sure that the webview loaded by the Jetpack mobile App doesn't contain any invalid section or links.

The proposed changes are:
* Hide the new header buttons and subtitles in mobile Jetpack
* Hide the Support button in the campaign details page
* Added a margin-top to the search component

Like in the previous version, I continue to hide the navigation bar.

## Testing Instructions

You need to test this in a browser with a specific user agent. In Chrome for example, in the device dimension dropdown, select Edit and then Add Custom.

Add this in the `User agent string` field: 
`Mozilla/5.0 (iPhone; CPU iPhone OS 12_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/16B91 wp-iphone/12.1`

* Load the live preview and navigate to Tools->Advertising
* In the URL, activate the feature flag `promote-post/redesign-i2` and refresh the page.
* Verify that the WebView version of the Posts page is displayed
**Note:** No header or navigation, and without the `View` link in the posts.

![Screenshot 2023-06-12 at 16 06 48](https://github.com/Automattic/wp-calypso/assets/1258162/d999536c-a14f-4ecd-91c4-95e022b10e35)

* Exit the mobile view mode and refresh the page
* Go to the Campaigns page and then select a campaign from the list to see the detail
* Go to mobile mode again, in the URL, activate the feature flag `promote-post/redesign-i2` and refresh the page
* Verify that the Campaign details page is displaying the WebView version of the page
**Note:** No header (only the Back button), and without the `Contact Support` link in the page.


![Screenshot 2023-06-12 at 16 32 41](https://github.com/Automattic/wp-calypso/assets/1258162/679346ec-f885-444c-aadf-6caa8538a0d1)

![Screenshot 2023-06-12 at 16 06 39](https://github.com/Automattic/wp-calypso/assets/1258162/40ba34c8-cf1e-4c9b-b88a-a78d95a90e16)

## Pre-merge Checklist
